pastastore/util.py

```diff
@@ -399,3 +400,232 @@ def copy_database(conn1, conn2, libraries: Optional[List[str]] = None,
+                    cov_Ab = ml.fit.pcov.loc[f"{sm_name}_A", f"{sm_name}_b"]
+                    gain_std = np.sqrt(sm.rfunc.variance_gain(
+                        A, b, var_A, var_b, cov_Ab))
+                    if gain_std is None:
+                        gain_std = np.nan
+                        check_gain_passed = pd.NA
+                    elif np.isnan(gain_std):
+                        check_gain_passed = pd.NA
+                    else:
+                        check_gain_passed = np.abs(gain) > 2 * gain_std
+                    checks.loc[f"gain > 2*std: {sm_name} (r=1)"] = (
+                        gain, 2 * gain_std, "m/(Mm3/yr)", check_gain_passed)
+
+                    for iw in range(sm.distances.index.size):
+                        p = sm.get_parameters(model=ml, istress=iw)
+                        gain = sm.rfunc.gain(p)
+                        gain_std = np.sqrt(
+                            sm.variance_gain(model=ml, istress=iw))
+                        if gain_std is None:
+                            gain_std = np.nan
+                            check_gain_passed = pd.NA
+                        elif np.isnan(gain_std):
+                            check_gain_passed = pd.NA
+                        else:
+                            check_gain_passed = np.abs(gain) > 2 * gain_std
+                        checks.loc[f"gain > 2*std: {sm_name}-{iw:02g}"] = (
+                            gain, 2 * gain_std, "m/(Mm3/yr)", check_gain_passed)
+                else:
+                    gain = ml.parameters.loc[f"{sm_name}_A", "optimal"]
+                    gain_std = ml.parameters.loc[f"{sm_name}_A", "stderr"]
+                    if gain_std is None:
+                        gain_std = np.nan
+                        check_gain_passed = pd.NA
+                    elif np.isnan(gain_std):
+                        check_gain_passed = pd.NA
+                    else:
+                        check_gain_passed = np.abs(gain) > 2 * gain_std
+                    check_gain_passed = np.abs(gain) > 2 * gain_std
+                    checks.loc[f"gain > 2*std: {sm_name}"] = (
+                        gain, 2 * gain_std, "m/(Mm3/yr)", check_gain_passed)
+
+        # Check 5 - Parameter Bounds
+        if check5:
+            upper, lower = ml._check_parameters_bounds()
+            for param in ml.parameters.index:
+                bounds = (ml.parameters.loc[param, "pmin"],
+                          ml.parameters.loc[param, "pmax"])
+                b = ~(upper.loc[param] or lower.loc[param])
+
+                checks.loc[f"Parameter bounds: {param}", :] = (
+                    ml.parameters.loc[param, "optimal"], bounds, "_", b)
+
+        if checks['check_passed'].all():
+            check_passed_models.append(ml.name)
+
+        if path:
+            checks.to_csv(f"{path}/checks_{ml.name}.csv",
+                          na_rep="NaN")
+
+    return check_passed_models
```

Comment:
instead of returning this list, wouldnt it be nicer to return a summary DataFrame with model names as index, check_passed columns for each check, and perhaps a last column containing whether all checks were passed by a model?

Reply:
I agree, I will transform it. However, most of the other Pastastore functions work with lists of models so that is something to keep in mind.

Reply:
True, but as long as deriving that list from the Pandas DataFrame is very straightforward I prefer returning the DataFrames as they contain more (useful) info.